nixos/modules/programs/nix-required-mounts.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.nix-required-mounts;
+  package = pkgs.nix-required-mounts;
+  overridenPackage = package.override { inherit (cfg) allowedPatterns; };
+
+  Mount = with lib;
+    types.submodule {
+      options.host = mkOption {
+        type = types.str;
+        description = "Host path to mount";
+      };
+      options.guest = mkOption {
+        type = types.str;
+        description = "Location in the sandbox to mount the host path at";
+      };
+    };
+  Pattern = with lib.types;
+    types.submodule ({ config, name, ... }: {
+      options.onFeatures = lib.mkOption {
+        type = listOf types.str;
+        description =
+          "Which requiredSystemFeatures should trigger relaxation of the sandbox";
+        default = [ name ];
+      };
+      options.paths = lib.mkOption {
+        type = listOf (oneOf [ path Mount ]);
+        description =
+          "A list of glob patterns, indicating which paths to expose to the sandbox";
+      };
+      options.unsafeFollowSymlinks = lib.mkEnableOption ''
+        Instructs the hook to mount the symlink targets as well, when any of
+        the `paths` contain symlinks. This may not work correctly with glob
+        patterns.
+      '';
+    });
+
+  driverPaths = [
+    # symlinks in /run/opengl-driver/lib:
+    pkgs.addOpenGLRunpath.driverLink
+
+    # the actual userspace drivers in /nix/store:
+    config.hardware.opengl.package
+  ] ++ config.hardware.opengl.extraPackages;
```

Comment:
It just occurred to me that I'm not explicitly mounting targets of the RUNPATHs in the userspace drivers. For instance, I cannot explain why the exact `libc` version linked to by `libcuda.so` happens to be available in the sandbox, but I suspect that's just a coincidence

Reply:
You probably need to use https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-exportReferencesGraph to get the complete closure of the driver runtime.